### PR TITLE
feat: implement multi cluster support

### PR DESCRIPTION
**Pull request by arielsepton**
This PR introduces multi-cluster support. 

## Changes
* Multi-Cluster Support: enable routing across different clusters.
* New Clusters Routes: Defined under `/clusters/:clusterName/namespaces/:namespaceName`,  providing access to:
    * Logs
    * Containers
    * Pods

### CODE REVIEW

File: src/routes/v1/logs_stream_test.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func Test_GetCappLogs(t *testing.T) {
 		"ShouldStreamLogsWithoutQueryParams": {
 			args: args{
 				token: "valid_token",
-				wsUrl: "/v1/logs/capp/" + testNamespaceGetCappLogs + "/" + testutils.CappName,
+				wsUrl: "/v1/namespaces/" + testNamespaceGetCappLogs + "/capp/" + testutils.CappName + "/logs",
```
Comment (fran2810) [inline src/routes/v1/logs_stream_test.go:44]: i think sprintf in this scenarios will be cleaner

File: test/e2e_tests/helper.go [hunk 2]
```diff
@@ -61,32 +62,23 @@ func compareResponses(response, expectedResponse map[string]interface{}) {
...
-	if expectedValue == nil {
-		Expect(value).To(BeNil())
+	if !expectedResponseHasError {
+		Expect(responseHasError).Should(Equal(false), fmt.Sprintf("unexpected error: %s", responseError))
 		return
 	}
 
-	Expect(value).To(Equal(expectedValue), "Value for key '%s' does not match expected value", key)
+	Expect(responseError).To(ContainSubstring(expectedError.(string)), "error %q does not contain expected error %q", responseError, expectedError)
+	delete(response, "error")
```
Comment (fran2810) [inline test/e2e_tests/helper.go:80]: can you explain me this lines and why it needed ?

File: src/middleware/context.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	"github.com/dana-team/platform-backend/src/customerrors"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetKubeClient retrieves the Kubernetes client from the gin.Context.
+func GetKubeClient(c *gin.Context) (kubernetes.Interface, error) {
+	kube, exists := c.Get(KubeClientCtxKey)
```
Comment (fran2810) [inline src/middleware/context.go:13]: dont see why its in middleware if we using it primarily in the routes  i think it could be in the routes

Comment (fran2810) [reply src/middleware/context.go:13]: except the last function AddErrorToContext

**Review by fran2810 — CHANGES_REQUESTED** [review]
love the changes to middleware

File: src/middleware/context.go (hunk 3 above)
Comment (arielsepton) [reply src/middleware/context.go:13]: These clients and the logger are set in the context by middleware, so it makes sense to have the retrieval functions in the middleware as well. 
`GetLogger`, `GetDynClient`, and `AddErrorToContext` are used within middlewares, so keeping them together in the middleware package maintains logical grouping.

File: Makefile [hunk 4]
```diff
@@ -112,10 +115,26 @@ env-file:
...
+uninstall-cluster-proxy-role:
+	kubectl delete -f $(CLUSTER_PROXY_RBAC)
+
+.PHONY: install-cluster-gateway-role
+install-cluster-gateway-role:
+	kubectl apply -f $(CLUSTER_GATEWAY_RBAC)
+
+.PHONY: uninstall-cluster-gateway-role
+uninstall-cluster-gateway-role:
+	kubectl delete -f $(CLUSTER_GATEWAY_RBAC)
```
Comment (mzeevi) [inline Makefile:137]: Do we need both?

Comment (arielsepton) [reply Makefile:137]: Yes, I added it after seeing this failure: [link](https://github.com/dana-team/platform-backend/actions/runs/10831513878/job/30058147698#step:15:894).

**Review by fran2810 — APPROVED** [review]
/lgtm

**Pull request merged**